File: Questions/CountPairs.java
import java.util.*;



public class TwoArrays {
    public static void main(String[] args) {
        Scanner sc = new Scanner (System.in);
        int n1 = sc.nextInt();
        int n2 = sc.nextInt();
        int[] arr1 = new int[n1];
        int [] arr2 = new int[n2];

        for(int i = 0; i < n1; i++) {
            arr1[i] = sc.nextInt();
        }
        for(int i = 0; i < n2; i++) {
            arr2[i] = sc.nextInt();
        }

        int target = sc.nextInt();

        System.out.println(countPairs(arr1, arr2, target));
    }

    public static boolean checkRemain(int[] arr2, int target){
        int tots = 0;
        for(int i = 0; i < arr2.length; i++){
            for(int j = i; j < arr2.length; j++){
                tots += arr2[j];
                if(tots == target){
                    return true;
                }else if(tots > target){
                    tots = 0;
                    break;
                }
            }
        }
        return false;
    }
    public static int countPairs(int[] arr1, int[] arr2, int target){
        int tots = 0;
        int ctr = 0;
        for(int i = 0; i < arr1.length; i++){
            for(int j = i; j < arr1.length; j++){
                tots += arr1[j];
                if (tots > target){
                    tots = 0;
                    break;
                }
                int rem = target - tots;
                if(checkRemain(arr2, rem)){
                    ctr++;
                }
            }
        }
        return ctr;
    }

}

File: Questions/Python/countPairs.py
arr1 = list(map(int, input().split()))
arr2 = list(map(int, input().split()))
target = int(input())


def checkRemain(arr2, remain):
    total = 0
    for i in range(len(arr2)):
        for j in range(i, len(arr2)):
            total += arr2[j]
            if total > remain:
                total = 0
                break
            elif total == remain:
                return True
    return False

def countPairs(arr1, arr2, target):
    total = 0
    count = 0
    for i in range(len(arr1)):
        for j in range(i, len(arr1)):
            total += arr1[j]
            if total > target:
                total = 0
                break
            remain = target - total
            if checkRemain(arr2, remain):
                count += 1
    return count

print(countPairs(arr1, arr2, target))